src/lib.rs

```diff
@@ -180,9 +180,18 @@ impl<F: PrimeField, PC: PolynomialCommitment<F>, D: Digest> Marlin<F, PC, D> {
 
         let prover_init_state = AHPForR1CS::prover_init(&index_pk.index, c, zk)?;
         let public_input = prover_init_state.public_input();
-        let mut fs_rng = PC::RandomOracle::from_seed(
-            &to_bytes![&Self::PROTOCOL_NAME, pc_pk.get_hash(), &index_pk.index_vk, &public_input].unwrap(),
-        );
+
+        let fs_rng_init_seed = {
+            let mut seed_builder = <PC::RandomOracle as FiatShamirRng>::Seed::new();
+            seed_builder.add_bytes(&Self::PROTOCOL_NAME).map_err(Error::from_pc_err)?;
+            seed_builder.add_bytes(&pc_pk.get_hash()).map_err(Error::from_pc_err)?;
+            // TODO: Shall we decompose this further when passing it to the seed builder ?
+            seed_builder.add_bytes(&index_pk.index_vk).map_err(Error::from_pc_err)?;
```

Comment:
Let us put a comment here: "As both vk and public input use constant length encoding of field elements, we can simply apply add_bytes to achieve a one-to-one serialization"